packages/core/src/api/Editor.ts

```diff
@@ -38,26 +43,77 @@ export class Editor {
     return blocks;
   }
 
+  /**
+   * Retrieves a snapshot of an existing block from the editor.
+   * @param block The identifier of an existing block that should be retrieved.
+   * @returns The block that matches the identifier, or undefined if no matching block was found.
+   */
+  public getBlock(block: BlockIdentifier): Block {
+    const id = typeof block === "string" ? block : block.id;
+    let newBlock: Block | undefined = undefined;
+
+    this.tiptapEditor.state.doc.firstChild!.descendants((node) => {
+      if (typeof newBlock !== "undefined") {
+        return false;
+      }
+
+      if (node.type.name !== "blockContainer" || node.attrs.id !== id) {
+        return true;
+      }
+
+      newBlock = nodeToBlock(node, this.blockCache);
+
+      return false;
+    });
+
+    if (typeof newBlock === "undefined") {
```

Comment:
I would return `undefined` here, I don't think it's intuitive to have a "get by id" function that throws an error if it's not found.

This means we do need to change the other functions (`updateBlock`, `replaceBlock`), because I do think those should throw an error if the block isn't found